src/borgstore/backends/s3.py

```diff
@@ -35,7 +35,7 @@ def get_s3_backend(url):
     """
     m = re.match(s3_regex, url, re.VERBOSE)
     if m:
-        s3type = "s3" if "s3:" in url else "b2"
+        s3type = "s3" if url.find("s3:") == 0 else "b2"
```

Comment:
Better than the original version. Is there a reason why you didn't use a named group within the regex match?

Besides that, one would normally use `.startswith(x)` for this instead of `.find(x) == 0`, but I still think the named group would be the best solution.

Reply:
Because I've given it a try initially when I tried to separate the types and it didn't work for whatever reason as such I decided not to invest more time in it.

Reply:
If it's such a blocker I'll try to find some time to figure out the regex named group, but for me it's not a priority.

Reply:
Made the change.